fms/models/__init__.py

```diff
@@ -65,6 +66,55 @@ def list_variants(architecture: str):
     return list(__models[architecture].keys())
 
 
+def __maybe_infer_model_variant(
+    architecture: Optional[str],
+    variant: Optional[str],
+    model_path: Optional[str],
+    source: Optional[str],
+    **kwargs,
+) -> Tuple[str, str, Dict[str, Any]]:
+    """Infer the model variant configuration from different sources, currently only supported sources are hf"""
+    extra_kwargs = kwargs
+    if architecture is None:
+        if variant is not None:
+            logging.warning(
+                f"architecture was set to None which implies the model configuration is being inferred (variant={variant} will be ignored)"
+            )
+
+        if model_path is None:
+            raise ValueError(
+                "the model_path parameter is required to be set in order to infer the model configuration when architecture=None"
+            )
+
+        logging.info(f"inferring model configuration from checkpoint at {model_path}")
+
+        if source is None:
```

Comment:
should we ignore 'source' under this case, since the 'inferred' model should always be HF? is there a possibility of an 'inferred' model config with weights sourced from elsewhere?

Reply:
The idea was if some other source can be inferred we can implement that when it comes along, for now its not necessary (just was future proofing)